backend/src/app.ts

```diff
@@ -1,25 +1,82 @@
 import express, { Application, Request, Response } from 'express'
 import { pg } from './db/database';
 
+const bcrypt = require('bcrypt');
 const dotenv = require('dotenv');
 dotenv.config();
 
 const app: Application = express()
 interface Tournament {
     id: string;
     name: string;
-    'joining-code': number;
+    player_code: string;
+    admin_code: string;
 }
 
-app.set('port', (process.env.PORT || 5000));
+app.use(express.json());
+app.use((req, res, next) => {
+    res.append('Access-Control-Allow-Origin', ['*']);
+    res.append('Access-Control-Allow-Methods', 'GET,PUT,POST,DELETE');
+    res.append('Access-Control-Allow-Headers', 'Content-Type');
+    next();
+});
+
+app.set('port', (process.env.PORT || 8000));
 
 app.get('/tournaments', async(req: Request, res: Response) => {
-    const query = pg.select("id", "name").from<Tournament>("tournament");
+    const query = pg.select("id", "name", "player_code", "admin_code").from<Tournament>("tournament");
     const tournaments = await query;
     
     res.json( { tournaments });
 });
 
+app.post('/tournaments', async(req: Request, res: Response) => {
+    const body = req.body;
+    const adminCode = body.admin_code;
+    const playerCode = body.player_code;
+
+    const adminHash = await bcrypt.hash(adminCode, 10);
+    const playerHash = await bcrypt.hash(playerCode, 10);
+
+    body.admin_code = adminHash;
+    body.player_code = playerHash;
+
+    for (let requiredParameter of ['name', 'player_code', 'admin_code']) {
+        if (!body[requiredParameter]) {
+          return res
+            .status(422)
+            .json({ error: `Expected format: { name: <String>, player_code: <String>, admin_code: <String> }. You're missing a "${requiredParameter}" property.` });
+        }
+    }
+    
+    try {
+    const tournamentIdArray = await pg("tournament").insert(body).returning("id").then(function (id) { return id; });
+
+    const tournamentId = tournamentIdArray[0];
+    body.id = tournamentId;
```

Comment:
Compare the POST response you have here to the one of the POST /customers endpoint on the fruit shop API: https://api.predic8.de/shop/docs#!/customers/postCustomers

Please return the full tournament object (except for the hashed passwords) back to the user, not just the ID.

Reply:
The endpoint already return the full tournament object. But I removed the hashed passwords out of the return.

Reply:
Whoops sorry. Must've misread the code. Good that you removed the hashed passwords. They're of no use to any user and are better kept private.